### PR TITLE
17258 - Adding inital api.

### PR DESCRIPTION
**Pull request by hfekete**
[Issue: 17258](/bcgov/entity/issues/17258)

- Adding initial API.

### CODE REVIEW

File: btr-api/src/btr_api/models/ownership_details.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    __tablename__ = "ownership_details"
+
+    business_identifier = Column(String(300), index=True, nullable=False)
+    control_type = Column(String(50), nullable=False)
+    control_percent = Column(Integer(), nullable=False)
+    additional_text = Column(String(2000), nullable=True)
+
+    # Relationships
+    # entity_id = db.Column('entity_id', db.Integer, db.ForeignKey('entity.id'))
```
Comment (kialj876) [inline btr-api/src/btr_api/models/ownership_details.py:20]: saving this for later? I think you can remove it and we'll add it when we need

File: btr-api/src/btr_api/models/base.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+from .db import db
+
+
+class BtrModelBase:
+    id = db.Column(db.Integer, primary_key=True)
+
+    created_by = Column(Integer)
+    updated_by = Column(Integer)
+    created_at = Column(DateTime(timezone=True), server_default=func.now())
+    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
```
Comment (kialj876) [inline btr-api/src/btr_api/models/base.py:13]: This sort of thing would already be tracked via the submission table right? That would be our ledger for tracking updates. Maybe instead we should add submission_id here instead or we can go without it?

File: btr-api/src/btr_api/models/ownership_details.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    person_id = db.Column('person_id', db.Integer, db.ForeignKey('persons.id'))
+    person = db.relationship('Person', backref=backref('filing_submitter', uselist=False), foreign_keys=[person_id])
```
Comment (kialj876) [inline btr-api/src/btr_api/models/ownership_details.py:23]: Can you rename the backref to 'person'?

File: btr-api/src/btr_api/services/person.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+
+class PersonService(object):
+    @staticmethod
+    def save_person_from_submission(submission_dict: dict) -> PersonModel | None:
+        if 'person' in submission_dict:
+            person_dict = submission_dict['person']
+            person: PersonModel = PersonSerializer.from_dict(person_dict)
+            return person
+        print('bla')
```
Comment (kialj876) [inline btr-api/src/btr_api/services/person.py:68]: forgot to remove this one

File: btr-api/src/btr_api/models/person.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    full_name = Column(String(300), index=True, nullable=True)
+    family_name = Column(String(100), nullable=True)
+    given_name = Column(String(100), nullable=True)
+    patronymic_name = Column(String(100), nullable=True)
+
+    date_of_birth = Column(DateTime(timezone=True))
+
+    is_permanent_resident = Column(Boolean(), default=False)
+    is_canadian_citizen = Column(Boolean(), default=False)
+    is_canadian_tax_resident = Column(Boolean(), default=False)
```
Comment (kialj876) [inline btr-api/src/btr_api/models/person.py:26]: if we don't add submission_id to the base model I think it should be added here so we don't lose the submission link to each record

**Review by kialj876 — COMMENTED** [review]
Nice work this looks good, I just have a few small comments

File: btr-api/src/btr_api/models/base.py (hunk 2 above)
Comment (hfekete) [reply btr-api/src/btr_api/models/base.py:13]: kept it, as we discussed to keep some tables separate from the submission

File: btr-api/src/btr_api/models/person.py (hunk 5 above)
Comment (hfekete) [reply btr-api/src/btr_api/models/person.py:26]: Updated relationships as we discussed offline.

**Review by kialj876 — APPROVED** [review]
Awesome 👍

**Pull request merged**